java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/MainActivity.java

```diff
@@ -0,0 +1,330 @@
+                    true);
+			
+			HealthVaultSettings settings = service.getSettings();
+			settings.setMasterAppId("c6ba979f-c342-4408-a2bc-0dfb43b2bf8d");
+			settings.setServiceUrl("https://platform.healthvault-ppe.com/platform/wildcat.ashx");
+			settings.setShellUrl("https://account.healthvault-ppe.com");
+			settings.setIsMultiInstanceAware(true);
+			settings.setIsMRA(true);
+			service.start(MainActivity.this, MainActivity.this);
+    	}
+		else {
+			Toast.makeText(
+	                MainActivity.this, 
+	                "App is already connected", 
+	                Toast.LENGTH_LONG).show();
+		}
+	}
+	
+	public void onListItemClick(ListView parent, View v, int position, long id) {
+        TextView item = (TextView) v;
+        Intent intent = null;
+        if (service.isAppConnected()) {
+	        switch(position) {
+	        case 0: 
+        		intent = new Intent(MainActivity.this, WeightActivity.class);
+	        }
+        }
+        
+        if(intent != null) {
+        	startActivity(intent);
+        }
+    }
+	
+	/*private void registerHandlers() {
+
+		// Connect button
+		Button connect = (Button) findViewById(R.id.connect);
+		connect.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View view) {
+            	doConnect();
+            }
+        });
+		
+		Button weightsBtn = (Button) findViewById(R.id.weights);
+		weightsBtn.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View view) {
+            	if (service.isAppConnected()) {
+            		Intent intent = new Intent(MainActivity.this, WeightActivity.class);
+                    startActivity(intent);
+            	}
+            }
+        });
+		
+		Button fileUploadBtn = (Button) findViewById(R.id.fileUpload);
+		fileUploadBtn.setOnClickListener(new View.OnClickListener() {
+            @SuppressWarnings("unchecked")
+			public void onClick(View view) {
+            	if (service.isAppConnected()) {
+            		String filename = writeFile();
+            		
+            		InputStream source;
+					try {
+						source = openFileInput(filename);
+						
+						File hvFile = new File();
+						hvFile.setName(filename);
+						
+						hvClient.start();
+	            		
+	            		hvClient.asyncRequest(hvFile.uploadAsync(((HVApp)getApplication()).getCurrentRecord(), null, source), 
+	            				new MainActivityCallback(MainActivityCallback.UpdateRecords));
+					} catch (Exception e) {
+						// TODO Auto-generated catch block
+						e.printStackTrace();
+					}
+            	}
+            }
+        });
+		
+		Button vocabTest = (Button) findViewById(R.id.vocabTest);
+		vocabTest.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View view) {
+            	Intent intent  = new Intent(MainActivity.this, VocabActivity.class);
+            	startActivity(intent);
+            }
+        });
+	}*/
+
+	private String writeFile() {
```

Comment:
More indentation issues, I won't keep adding the comments because it looks like it is throughout all the files, but we need to do a pass to correct this.